packages/react-form/src/hooks/list/baselist.ts

```diff
@@ -86,5 +93,13 @@ export function useBaseList<Item extends object>(
     });
   }, [state.list, handlers]);
 
-  return {fields, dispatch};
+  const fieldsDirty = fields.some(field => {
+    const keys: [keyof Item] = Object.keys(field) as [keyof Item];
+
+    return keys.some(key => field[key].dirty);
+  });
+
+  const dirty = fieldsDirty || state.list.length !== state.initial.length;
```

Comment:
We could have an early return if `state.list.length !== state.initial.length`, this would avoid to loop though each item for no purpose.